core-processor/src/configs.rs

```diff
@@ -125,3 +131,53 @@ impl ExecutionSettings {
         self.allocations_config.load_page_cost
     }
 }
+
+/// Stable parameters for the whole block across processing runs
+#[derive(Clone)]
+pub struct BlockConfig {
+    /// Block info
+    pub block_info: BlockInfo,
+    /// Allocations config
+    pub allocations_config: AllocationsConfig,
+    /// Existential deposit
+    pub existential_deposit: u128,
+    /// Gas allowance
+    pub gas_allowance: u64,
+    /// Outgoing limit
+    pub outgoing_limit: u32,
+    /// Host function weights
+    pub host_fn_weights: HostFnWeights,
+    /// Forbidden functions
+    pub forbidden_funcs: BTreeSet<&'static str>,
+    /// Mailbox threshold
+    pub mailbox_threshold: u64,
+}
+
+impl Default for BlockConfig {
+    fn default() -> Self {
+        Self {
+            block_info: Default::default(),
+            allocations_config: Default::default(),
+            existential_deposit: EXISTENTIAL_DEPOSIT,
+            gas_allowance: u64::MAX,
+            outgoing_limit: OUTGOING_LIMIT,
+            host_fn_weights: Default::default(),
+            forbidden_funcs: Default::default(),
+            mailbox_threshold: MAILBOX_THRESHOLD,
+        }
+    }
+}
+
+/// Unstable parameters for message execution across processing runs
+#[derive(Clone)]
+pub struct MessageExecutionConfig {
```

Comment:
I'd also rename this to execution context/data
than we'll  have more logical:
`fn process(to_execute: ExecutionData, config: &BlockConfig)`